app/src/projects.js

```diff
@@ -54,9 +54,42 @@ router.use(auth.middleware.credentials);
 /**
  * Show a list of all the users projects.
  */
-router.get('/', (req, res) => {
-    db.get().collection('projects').find({ "owner": req.cookies.u, $or: [{ hidden: { $exists: false } }, { hidden: false }] }).project({ title: true, id: true }).toArray((err, projects) => {
-        ejs.renderFile(`app/views/projects.ejs`, { projects_data: projects }, {}, (err, str) => {
+router.get('/', async (req, res) => {
+
+    // Create array of last modified timeStamps for each project
+
+    let timeStamps = [];
+    let timeMap = {};
+    findMTime = db.get().collection('project_data').find({});
+    await findMTime.forEach(
+        doc => {
+            timeStamps.push(doc._m.mtime);
+            let forConversion = new Date(doc._m.mtime).toString();
+            let splitter = forConversion.split("T");
+            // console.log("Last edited on: " + splitter[0]);
+            timeMap[doc._id] = splitter[0];
+        }
+    );
```

Comment:
Isn't this going to retrieve the entire `project_data` collection due to line 63? For large database this would take a while to process or hog system memory I think.